### DISCUSSION TITLE
Workaround for flutter 3.22.0 colorscheme issue

### DISCUSSION THREAD

**Discussion by realth000** [rydmike/flex_color_scheme#232] [Q&A] [answered]
I'm using the latest version of flex_color_scheme (7.3.1).

After upgrade to Flutter 3.22.0, some background part in widgets (such as `Card`) become brighter (not that much primary color as before).

Maybe [flutter/issue148359](https://github.com/flutter/flutter/issues/148359) is related?

It appears when only setting `scheme`:

```dart
final colorScheme = FlexThemeData.light(
  scheme: FlexScheme.bahamaBlue,
);
```

Disappears when I override color roles in `FlexThemeDta.light()`:

```dart
final seedScheme = ColorScheme.fromSeed(seedColor: seedColor);
final fixedColorScheme = FlexThemeData.light(
  primary: seedScheme?.primary,
  onPrimary: seedScheme?.onPrimary,
  primaryContainer: seedScheme?.primaryContainer,
  onPrimaryContainer: seedScheme?.onPrimaryContainer,
  // other color roles.
);
```

I don't know how to change it back, maybe specify `CardTheme`? But where I define the colorscheme is outside of `BuildContext`, const color value can not work with all primary colors.

![cardtheme](https://github.com/flutter/flutter/assets/59462104/145eddad-82c4-48eb-8118-d80018becb2e)

Is there any workaround to fix background color when only using `scheme`?

**Comment by rydmike** [comment, marked as answer]
FlexColorScheme 7.3.1 and earlier version is not fully compatible with Flutter 3.22, since Flutter 3.22 contains many breaking changes in how it treats and handles surface colors.

FlexColorScheme 8.0.0, that will also contain breaking changes in order to adjust for all the breaking changes in Flutter 3.22 will resolve these breaking style changes in the manner intended by Flutter and Material design, but they will still not be identical to the styles prior to Flutter 3.22. 

Yes, the less colored surface colors on e.g. elevated Card are a part of breaking style changes in Material-3 design. But some style changes you observe on some surfaces, if using FCS 7.3.1 or earlier with Flutter 3.22, may also be due to different usage of surface color mappings between the style breaking Flutter version 3.22 and FCS 7.3.1.  


https://docs.flexcolorscheme.com/
![Screenshot 2024-05-19 at 02 44 26](https://github.com/rydmike/flex_color_scheme/assets/39990307/50410197-01fd-4bf1-86fb-bb0ca8b654bb)

**Reply by realth000** [reply]
Thanks! I will wait for 8.0.0 version no matter how long it will take, thanks for this great package.

**Reply by rydmike** [reply]
I will publish dev releases 8.0.0-dev.1 (...n) first when all the basics are OK, but it won't have its companion Themes Playground ready during dev releases of the package. I won't call it a stable 8.0.0 until Themes Playground is also updated to support Flutter 3.22. The bulk of the work is in updated Playground support and all the test in FCS as it aims for 100% coverage. The dev releases will skimp on some new tests too, to get something usable out faster for those that want to live on the edge.

**Reply by rydmike** [reply]
**Some more info** 

Flutter 3.22 broke +100 tests and had +2000 deprecation warnings/hints in FlexColorScheme package and Themes Playground together. It will all be sorted, but the breakage is substantial and it will take some time before a compatible version that lives up to past high quality can be released. Plus I will be traveling the next few days, so that will slow down progress a bit :( 

You can follow progress in my version 8.0.0 change log. I update it as I work on the 1st dev release of 8.0.0:
* https://github.com/rydmike/flex_color_scheme/blob/version-8-0-0/CHANGELOG.md

**Reply by realth000** [reply]
Have a good time

Maybe we can have more pre-defined color schemes about different travelling places in future version (far away from this discussion)

**Reply by rydmike** [reply]
For info FCS 8.0.0 is out as early access as 8.0.0-dev.1

<img width="1265" alt="Screenshot 2024-10-15 at 4 12 50" src="https://github.com/user-attachments/assets/cf4c6d79-242e-453c-9762-cf967c92ce2c">


**FlexColorScheme** package dev release 8.0.0-dev.1 has been published. This is a BIG update with breaking changes and many new features. It has its own dev pre-release of the [**Themes Playground**](https://rydmike.com/flexcolorscheme/themesplayground-dev) as well, feel free to take a look. No new API changes are planned for the stable 8.0.0 release from this dev.1 release.

We would appreciate any feedback about found issues from early adopters. 

* [Doc site has some pre-info](https://docs.flexcolorscheme.com/)
* [Dev release of PKG here](https://pub.dev/packages/flex_color_scheme/versions/8.0.0-dev.1)
* [The PKG changelog has the most detailed info](https://pub.dev/packages/flex_color_scheme/versions/8.0.0-dev.1/changelog)

**FOR INFO** there will be no new features or changes in it compared to this one. Only any issues found from the quality assurance will be addressed. The stable release will also include many doc updates, but docs site will continue to be edited and revised AFTER the 8.0.0 stable release.

**Reply by realth000** [reply]
yeah I saw the annoucement yesterday, a so long changlog.

I will try this pre-release tonight, thanks again for this excellent package.

> Btw there's a coincedense that now I'm working on a app also in pre-release stage, seems I can rework my UI with these new features, thank you!

**Reply by rydmike** [reply]
And now it is released as 8.0.0 stable

- https://github.com/rydmike/flex_color_scheme
- https://rydmike.com/flexcolorscheme/themesplayground-latest/
- https://docs.flexcolorscheme.com/whats_new
